control/celeryapp.py

```diff
@@ -59,6 +66,28 @@ def deliverable_route(veda_id, encode_profile):
     veda_deliver.run()
 
 
+@app.task(name='ingest_worker')
+def ingest(studio_upload_id, video_metadata, course_id, course_hex):
+    """
+    Ingest a video in s3.
+    Arguments:
+        - studio_upload_id: a valid studio upload ID
+        - video_metadata: video metadata
+        - course_id: course id identifying a course run
+        - course_hex: studio_hex identifying course runs
+    """
+    course = get_or_create_course(
+        course_id,
+        course_hex=course_hex
+    )
+    veda_ingest = VedaIngest(
+        course_object=course,
+        video_proto=VideoProto(**video_metadata),
```

Comment:
You need to get the video_metadata from the database and S3 key object

Reply:
👍